### ISSUE TITLE
Manually check for plugin/theme updates

### ISSUE THREAD

**Issue by kontrollanten** [Chocobozzz/PeerTube#4735]
**Describe the problem to be solved**
Sometimes a plugin or theme needs to be updated immediately. Today it can take up to 24 hours before the PeerTube plugin registry gets updated.

**Describe the solution you would like:**
Possibility for an admin to manually check for updates.

**Describe alternatives you have considered**  
Add a button to the installed plugins page that makes a request to the official npm registry to see if there's any update available.

**Comment by Utopiah** [comment]
Indeed, one suggestion I had in mind was to use the search component to bypass the registry. For example one would search for https://www.npmjs.com/package/peertube-plugin-example and instead of searching in the registry it would hit npm directly. This would be a convenient way to both install a new plugin that has just been published or force an update before the 24h period.

**Comment by emansom** [comment]
> Indeed, one suggestion I had in mind was to use the search component to bypass the registry. For example one would search for https://www.npmjs.com/package/peertube-plugin-example and instead of searching in the registry it would hit npm directly. This would be a convenient way to both install a new plugin that has just been published or force an update before the 24h period.

That'd bypass quality control and manual checks for malware.

I rather have a slow update process where someone at Framasoft verifies stuff and not get compromised, than random developer X selling his npm module, politically motivated malware injection or flat out credential leak and get compromised.

**Comment by ROBERT-MCDOWELL** [comment]
why not to automate Q.O.C like moodle? https://docs.moodle.org/dev/Plugin_validation

**Comment by Utopiah** [comment]
Indeed @emansom but I believe if an administrator makes the decision of pasting the URL of an NPM project they are aware of the risk. This is not the typical process because for that the system already in place works.
